hadoop-hdds/common/src/main/java/org/apache/hadoop/hdds/scm/net/NetUtils.java

```diff
@@ -149,4 +149,28 @@ public static List<Node> getAncestorList(NetworkTopology topology,
     }
     return ancestorList;
   }
+
+  /**
+   * Ensure {@link NetConstants#PATH_SEPARATOR_STR} is added to the suffix of
+   * the path.
+   * @param path path to add suffix
+   * @return the normalised path
+   * If <i>path</i>is empty, then {@link NetConstants#ROOT} is returned
+   */
+  public static String addSuffix(String path) {
+    if (path == null) {
+      return null;
+    }
```

Comment:
Why are you returning ``null`` here? Won't this lead to NPE in the isDecendent(String ..) & isAncesstor(String ..) methods. 
For the same methods with argument as Node, if node is ``null``, you return false, but for the methods with argument as string you would return NPE, if the path is ``null``

Just FYI. In normalize method, it returns root in case of ``null``

```
  public static String normalize(String path) {
    if (path == null) {
      return NetConstants.ROOT;
    }
```

Reply:
I think it's a little different here. This method shouldn't change the fact that it's a null. If we return "/" for null strings, `isDescentof(null)` will always return true, which is not what we expected.

Reply:
ok